File: ratings/src/main/java/com/cinelist/ms/ratings/config/FeignClientConfiguration.java
package com.cinelist.ms.ratings.config;

import com.cinelist.ms.ratings.handlers.RetrieveMessageErrorDecoder;
import feign.codec.ErrorDecoder;
import org.springframework.context.annotation.Bean;

public class FeignClientConfiguration {

    @Bean
    public ErrorDecoder errorDecoder() {
        return new RetrieveMessageErrorDecoder();
    }
}

File: accounts/src/main/kotlin/com/cinelits/ms/accounts/config/FeignClientConfiguration.kt
package com.cinelits.ms.accounts.config

import com.cinelits.ms.accounts.handlers.RetrieveMessageErrorDecoder
import org.springframework.context.annotation.Bean

class FeignClientConfiguration {

    @Bean
    fun errorDecoder(): RetrieveMessageErrorDecoder {
        return RetrieveMessageErrorDecoder()
    }
}